### PR TITLE
Protect FindModuleBy functions with locks

### PR DESCRIPTION
**Pull request by kimmidi**
When used from multiple goroutines, the byNS and byPrefix maps can cause
concurrent write panics.

### CODE REVIEW

File: pkg/yang/modules.go [hunk 1]
```diff
@@ -229,6 +236,9 @@ func (ms *Modules) FindModuleByNamespace(ns string) (*Module, error) {
 // FindModuleByPrefix either returns the Module specified by prefix or returns
 // an error.
 func (ms *Modules) FindModuleByPrefix(prefix string) (*Module, error) {
+	ms.mu.Lock()
+	defer ms.mu.Unlock()
+
```
Comment (wenovus) [inline pkg/yang/modules.go:241]: This function (along with `byPrefix`) has been removed as of v0.3.0 (a backwards-incompatible version increment). Please use the function `FindModuleByPrefix` instead.

File: pkg/yang/modules.go [hunk 2]
```diff
@@ -200,6 +204,9 @@ func (ms *Modules) FindModule(n Node) *Module {
 // FindModuleByNamespace either returns the Module specified by the namespace
 // or returns an error.
 func (ms *Modules) FindModuleByNamespace(ns string) (*Module, error) {
+	ms.mu.Lock()
+	defer ms.mu.Unlock()
```
Comment (wenovus) [inline pkg/yang/modules.go:208]: Could you add a comment here that this is used to protect `byNS` from a race condition?

File: pkg/yang/modules.go (hunk 1 above)
Comment (kimmidi) [reply pkg/yang/modules.go:241]: Sorry, did you mean to suggest a different FindModule* function ?

Comment (wenovus) [reply pkg/yang/modules.go:241]: Yes this method no longer exists in v0.3.0. There is a *function* of the same name in node.go

**Pull request merged**